PWGHF/Tasks/taskMcEfficiencyToXiPi.cxx

```diff
@@ -212,13 +213,15 @@ struct HfTaskMcEfficiencyToXiPi {
 
     for (const auto& mcParticle : genParticles) {
 
-      // accept only mc particles coming from bc that are far away from TF border and ITSROFrame
-      if (rejGenTFAndITSROFBorders) {
-        auto coll = mcParticle.template mcCollision_as<aod::McCollisions>();
-        auto bc = coll.template bc_as<BCsInfo>();
-        if (!bc.selection_bit(o2::aod::evsel::kNoITSROFrameBorder) || !bc.selection_bit(o2::aod::evsel::kNoTimeFrameBorder)) {
-          continue;
-        }
+      auto coll = mcParticle.template mcCollision_as<aod::McCollisions>();
+      auto bc = coll.template bc_as<BCsInfo>();
+
+      // accept only mc particles coming from bc that are far away from TF border and/or ITSROFrame
+      if (rejGenTFBorders && !bc.selection_bit(o2::aod::evsel::kNoTimeFrameBorder)) {
+        continue;
+      }
+      if (rejGenITSROFBorders && !bc.selection_bit(o2::aod::evsel::kNoITSROFrameBorder)) {
+        continue;
```

Comment:
You could consider to use the event selection utility recently implemented for MC (also in a future PR)
https://github.com/AliceO2Group/O2Physics/blob/23af5c7dce65a351c40a42f7705fc701961170e3/PWGHF/Utils/utilsEvSelHf.h#L219

Reply:
I implemented it in this PR